### ISSUE TITLE
Getting unexpected keyword argument for to and reply_to

### ISSUE THREAD

**Issue by nickjj** [waynerv/flask-mailman#18]
Hey,

I'm noticing that when I send an email with: `mail.send_mail(**kwargs)` and kwargs has either or both of `to` / `reply_to` it says they are not allowed keyword arguments.

The value of `kwargs` right before calling `send_email` is:  
`{'subject': 'Just testing', 'to': ['redacted@example.com'], 'reply_to': 'someone@example.com'}`

It fails with:

```python
  return mail.send_mail(**kwargs)
TypeError: send_mail() got an unexpected keyword argument 'to'
```

And if I remove `to` and leave `reply_to` the error will say unexpected keyword argument `reply_to`. There's lots of mentions in the docs saying this should be allowed. 

It seems to be ok with the `subject` keyword argument, that never throws an error.

Any thoughts on what's happening?

**Comment by waynerv** [comment]
Hi, please note that `send_mail` and `EmailMessage` have different interface and parameter naming.
- `send_mail` accept `subject`, `message`, `from_email`, `recipient_list`, etc(but no `reply_to`).
- `EmailMessage` take `subject`, `body`, `from_email`, `to`, `bcc`, etc.

`send_mail` is just a tool function provided at your convenience, I didn't change the naming of its parameters to be compatible with Django, you can simply wrap one yourself.

If the docs is not clear enough, just let me know and I will find a way to improve it.

**Comment by nickjj** [comment]
Oh. I think the docs mentioned that `send_email` can be used in most cases, but if it doesn't support `reply_to` I'm not sure if it can be used since setting that value is important.

Also based on the docs for `send_mail`'s it says `Each member of recipient_list will see the other recipients in the “To:” field of the email message.` which I guess is different behavior than `to`?

**Comment by waynerv** [comment]
That's just a bonus tip for [difference bettween To Cc and Bcc](https://blog.thedigitalgroup.com/to-vs-cc-vs-bcc-how-to-use-them-correctly#:~:text=The%20TO%20and%20CC%20fields,your%20recipients%20view%20the%20email.&text=The%20BCC%20field%20is%20used,you%20have%20sent%20them%20to.), they have exactly the same behavior.

I think in your use case you should just use `EmailMessage`, It gives you complete ability.

I will update the document to make it clearer and thank you for your suggestions.

**Comment by nickjj** [comment]
Thanks, previously I followed the docs which had this:

```python
from flask import Flask
from flask_mailman import Mail

mail = Mail()

app = Flask(__name__)
mail.init_app(app)
```

And then in my app I took that same `mail = Mail()` instance that was created and used `mail.send_mail(...)`.

If I'm going to be using EmailMessage does that mean I should never use that instance of `mail`, and instead create a new instance of `EmailMessage` every time I send a mail instead of being able to reuse the same `mail` instance that was created when I started the app?

**Comment by nickjj** [comment]
Also if using `EmailMessage` how do you send both a txt and html email? With `send_mail` it had separate arguments for it.

But the docs for `EmailMessage` say you can only send text, however you can set the `content_subtype` to be "html" and then pass in HTML to the `body` instead of text. But it's not clear on what happens when the recipient can't receive HTML, how do you pass in both text and HTML like `send_email`? I didn't see any tests for this.

Basically I want to create both an html and txt template and send both and let the client pick whatever they want to use. This seems to be the behavior you would get with `send_mail`.

**Comment by waynerv** [comment]
The instantiating process of `Mail` is to get the configuration from the flask `app` and store it as state, which is the way the Flask documentation requires, so almost every Flask extensions must do this.

When you call `EmailMessage().send()`, its behind-the-scenes mechanism is also to automatically use the configuration stored in 
existing `mail` instance and will raise an error if `Mail` has not been instantiated.

**Comment by waynerv** [comment]
You can simply wrap a [`EmailMultiAlternatives`](https://www.waynerv.com/flask-mailman/#sending-alternative-content-types):
```
def send_mail(subject, message, from_email, recipient_list,html_message=None):
        msg= EmailMultiAlternatives(subject, message, from_email, recipient_list)
        if html_message:
            mail.attach_alternative(html_message, 'text/html')

        return msg.send()
```

**Comment by waynerv** [comment]
Close it.
